### PR TITLE
[9.0][MIG]web_ir_actions_act_window_message: Migration to V9

### PR DESCRIPTION
**Pull request by zakiuu**
* Upgrade  module web_ir_actions_act_window_message version to 9.0.
* Align file header to the OCA Guidelines.
* Migrate Javascript file to the V9 API.
* Refresh the view before closing the dialog when clicking
on close button

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please squash "OCA Transbot" commits together (https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests).

There's also an in progress migration to 10.0 in #605 that both can check for finding common solutions.

**Comment by zakiuu** [comment]
@pedrobaeza done

**Comment by pedrobaeza** [comment]
Thanks!

File: web_ir_actions_act_window_message/README.rst
Comment (sbidoul) [inline web_ir_actions_act_window_message/README.rst]: Would it make sense to add a `close` button type so the developer can control the position of the close button?

Comment (sbidoul) [reply web_ir_actions_act_window_message/README.rst]: Document the use of `oe_link_class` in button definition?

File: web_ir_actions_act_window_message/README.rst
Comment (sbidoul) [inline web_ir_actions_act_window_message/README.rst:36]: I think the terminology for such button in odoo views is `object`? Perhaps it's better to keep the same terminology.

**Comment by sbidoul** [comment]
A few comments about the API. I can't say much about the javascript.

File: web_ir_actions_act_window_message/README.rst
Comment (sbidoul) [inline web_ir_actions_act_window_message/README.rst]: I think the terminology for such button in odoo views is `object`? Perhaps it's better to keep the same terminology.

Comment (zakiuu) [reply web_ir_actions_act_window_message/README.rst:36]: @sbidoul I had the same opinion but I didn't want to change for compatibility if others have used the module

Comment (sbidoul) [reply web_ir_actions_act_window_message/README.rst:36]: Ah, sorry, I did not notice these had been implemented before by @hbrunn. So ignore these comments.

Comment (hbrunn) [reply web_ir_actions_act_window_message/README.rst:36]: I also considered that, but decided against it in the end because this is sufficiently different from view buttons, and I hate having to type `object` for calling a method every single time I put a button on the form ;-)

File: web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js
Comment (hbrunn) [inline web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js:29]: good one!

**Review by hbrunn — APPROVED** [review]
seems fine to me, although I think it's a bad habit to change coding style if there's no convention dictating the change. (I mean the braces placement - I happen to prefer BSD style for non-inline functions, but I wouldn't start changing this for no reason in a piece of code following a different paradigm. You make it worse by not changing this everywhere, so the code as a whole doesn't follow any recognizable convention)

**Comment by zakiuu** [comment]
Thanks @hbrunn, I really didn't take that into consideration

File: web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js
Comment (zakiuu) [inline web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js]: @hbrunn what do you think?

Comment (hbrunn) [reply web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js]: still approve, as I did before ;-) Or what specifically do you mean?

**Review by benwillig — APPROVED** [review]
Tested on runbot and works as expected.

File: web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js
Comment (benwillig) [inline web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js]: @zakiuu I just figured out that Odoo uses "classes" attribute instead of "oe_link_class" for dialog's buttons. "oe_highlight" class doesn't exists since 9.0.

```javascript
    set_buttons: function(buttons) {
        var self = this;

        self.$footer.empty();

        _.each(buttons, function(b) {
            var text = b.text || "";
            var classes = b.classes || ((buttons.length === 1)? "btn-primary" : "btn-default");
            var $b = $(QWeb.render('WidgetButton', { widget : { string: text, node: { attrs: {'class': classes} }}}));
```

**Review by benwillig — CHANGES_REQUESTED** [review]

File: web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js
Comment (zakiuu) [reply web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js]: @benwillig fixed

File: web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js
Comment (zakiuu) [inline web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js]: @hbrunn I've tried to respect the coding style u've put

File: web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js
Comment (benwillig) [inline web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js:44]: @zakiuu I think it would be a good idea to wrap line breaks.

```javascript
$content: $('<div>', {
        text: action.message,
        css: {
              'white-space': 'pre-line',
        }
}),
```

**Review by benwillig — CHANGES_REQUESTED** [review]

File: web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js
Comment (zakiuu) [reply web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js:44]: @benwillig thanks a lot

**Review by sbidoul — APPROVED** [review]
Fine with me. @hbrunn you may want to have a quick look at the last 3 commits. Then we'll migrate to 10.0.

**Comment by zakiuu** [comment]
@benwillig can you please approve the PR if everything is ok

**Review by benwillig — APPROVED** [review]

**Comment by sbidoul** [comment]
Works well. I merge this one. @hbrunn shout if you see potential issue with the css classes changes in the https://github.com/OCA/web/pull/611/commits/311e60d82f6ec72a1d9dbd37c97dce867b4e2a4d or 58fc2a53ca20b343607b26b36f54b3e3638f7a74.

**Comment by sbidoul** [comment]
@benwillig can you do the PR for 10.0?

**Comment by benwillig** [comment]
@sbidoul PR created #640

**Pull request merged**
